scripts/cplane/cluster_init.sh

```diff
@@ -0,0 +1,351 @@
+        echo -n "🔍 Verify Flannel Service running"
+        if [ ${elapsed} -gt 0 ]; then
+            echo " (trying for $((timeout - elapsed)) more seconds)"
+        else
+            echo ""
+        fi
+
+        # If Flannel is running correctly then it will report back the pods list
+        flannel_state=$(kubectl get pods -n kube-flannel -o json 2>/dev/null | jq -r '.items[] | select(.metadata.name | startswith("kube-flannel-")) | .status.phase')
+
+        elapsed=$((elapsed + interval))
+        if [ ${flannel_state} == "Running" ]; then
+            break
+        elif [ ${elapsed} -lt ${timeout} ]; then
+            sleep ${interval}
+        fi
+    done
+
+    if [ ${flannel_state} == "Running" ]; then
+        echo "✅ Flannel CNI Service is running"
+    else
+        echo "❌ Timed out during check - Flannel CNI service is NOT running"
+        exit 1
+    fi
+}
+
+
+# Install the Weave CNI (Container Network Interface)
+# NOTE: Weave Project was Shuttered in June 2024 and no longer supported
+function install_weave_cni {
+    echo "🚜  Install Weave CNI Service"
+    echo "--------------------------------------"
+    kubectl apply -f https://github.com/weaveworks/weave/releases/download/v2.8.1/weave-daemonset-k8s.yaml --validate=false
+    echo "--------------------------------------"
+}
+
+
+# Verify the Weave CNI Installation
+function verify_weave_cni {
+    echo "🔍 Verify Weave CNI Service Install"
+
+    # Verify that CNI service is running
+    timeout=60 # 2 minutes = 120 seconds
+    interval=10 # check every 10 seconds
+    elapsed=0
+
+    # Repeat loop until time runs out
+    while [ ${elapsed} -lt ${timeout} ]; do
+        echo -n "🔍 Verify Weave Service running"
+        if [ ${elapsed} -gt 0 ]; then
+            echo " (trying for $((timeout - elapsed)) more seconds)"
+        else
+            echo ""
+        fi
+
+        # If Weave is running correctly then it will report back the pods list
+        weave_count=$(kubectl get pods -n kube-system -o json 2>/dev/null | jq -r '.items[] | select(.metadata.name | startswith("weave-net-")) | .metadata.name' | wc -l)
+        elapsed=$((elapsed + interval))
+        if [ ${weave_count} -gt 0 ]; then
+            break
+        elif [ ${elapsed} -lt ${timeout} ]; then
+            sleep ${interval}
+        fi
+    done
+
+    if [ ${weave_count} -gt 0 ]; then
+        echo "✅ Weave CNI Service is running"
+    else
+        echo "❌ Timed out during check - Weave CNI service is NOT running"
+        exit 1
+    fi
+}
+
+
+# Install the Rancher LocalPath StorageClass implementation
+function install_localpath_storageclass {
+    echo "🚜  Install Rancher 'local-path' StorageClass"
+    echo "--------------------------------------"
+    kubectl apply -f https://raw.githubusercontent.com/rancher/local-path-provisioner/v0.0.32/deploy/local-path-storage.yaml --validate=false
+    echo "--------------------------------------"
+}
+
+
+#
+# Main Execution Loop
```

Comment:
Each function above is called here in one centralized area. This approach to script writing is similar to most procedureal languages and allows for clarity and debugging